src/services/db.js

```diff
@@ -82,40 +98,42 @@ export default class Database {
     return this.db.userTricks.clear();
   };
 
-  // populate the databes with tricks from the predefinedTricks.js
-  populateTricks = () => {
-    return this.db.predefinedTricks.clear().then(() => {
-      console.log("updating predefinedTricks")
-      const trickList = Papa.parse(predefinedTricks, {dynamicTyping: true}).data;
-
-      // this uses the labels of the csv but, also adds an id
-      const header = trickList.shift().concat(["stickFrequency"]);
-
-      const tricks = trickList.map(trick => {
-        // add 0 for stickFrequency
-        trick.push(0);
-        // make key value pairs
-        return Object.assign.apply({},
-          header.map((v,i) => {
+  // populate the database with tricks from the predefinedTricks.js
+  populateTricks = async () => {
+    await this.db.predefinedTricks.clear()
+
+    const trickList = Papa.parse(predefinedTricks, {dynamicTyping: true}).data;
+
+    // this uses the labels of the csv but, also adds an id
+    const header = trickList.shift().concat(["stickFrequency"]);
+
+    const tricks = trickList.map(trick => {
+      // add 0 for stickFrequency
+      trick.push(0);
+      // make key value pairs
+      return Object.assign.apply({},
+          header.map((v, i) => {
             return ({[v]: trick[i]})
```

Comment:
can be shortened to
```js
header.map( (v,i) => ({[v]: trick[i]})
```
although I am wondering why you chose this syntax over a simple
```js
const tricksLookup = {};
header.forEach((entry, i) => {
  tricksLookup[v] = trick[i];
}
return tricksLookup;
```
more readable imo.

Reply:
I didn't actually write this code but just removed tabs. But while I'm at it I'll implement the former suggestion.